### PR TITLE
Verify member variable is public before mocking

### PR DESCRIPTION
**Pull request by Dilhasha**
## Purpose
$subject
Fixes https://github.com/ballerina-platform/ballerina-lang/issues/30893

## Approach
check the access modifiers before allowing to stub member variables allowing to mock only public member variables

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: tests/testerina-integration-test/src/test/java/org/ballerinalang/testerina/test/MockTest.java
Comment (Thevakumar-Luheerathan) [inline tests/testerina-integration-test/src/test/java/org/ballerinalang/testerina/test/MockTest.java:177]: Having `\n\t` causes issues in actual ground truth outputs in windows. A corrupted output is generated. I have added a fix here(https://github.com/ballerina-platform/ballerina-lang/pull/40241). Can we wait until that PR is merged.

**Review by azinneera — APPROVED** [review]

**Pull request merged**
